framework/build.mk

```diff
@@ -70,7 +70,7 @@ unity_files::
 pcre%.$(obj-suffix) : pcre%.cc
 	@echo "MOOSE Compiling C++ (in "$(METHOD)" mode) "$<"..."
 	@$(libmesh_LIBTOOL) --tag=CXX $(LIBTOOLFLAGS) --mode=compile --quiet \
-          $(libmesh_CXX) $(libmesh_CPPFLAGS) $(ADDITIONAL_CPPFLAGS) $(libmesh_CXXFLAGS) $(app_INCLUDES) $(libmesh_INCLUDE) -DHAVE_CONFIG_H -MMD -MP -MF $@.d -MT $@ -c $< -o $@
+          $(libmesh_CXX) $(libmesh_CPPFLAGS) $(ADDITIONAL_CPPFLAGS) $(libmesh_CXXFLAGS) $(app_INCLUDES) $(libmesh_INCLUDE) -w -DHAVE_CONFIG_H -MMD -MP -MF $@.d -MT $@ -c $< -o $@
```

Comment:
Is this -w supposed to be here? I'm not familiar with that flag and I couldn't find any hits for it off hand.

Reply:
Yes, `-w`  suppresses **all** warnings. It works with gcc, clang, and intel

Reply:
OK - sounds good!